apps/server/test/lexical/server/code_intelligence/completion/translations/struct_field_test.exs

```diff
@@ -87,4 +87,78 @@ defmodule Lexical.Server.CodeIntelligence.Completion.Translations.StructFieldTes
     assert completion.detail == "first_name"
     assert completion.label == "first_name"
   end
+
+  describe "in struct arguments" do
+    def my_module(text) do
+      """
+      defmodule MyModule do
+        #{text}
+      end
+      """
+    end
+
+    test "should complete when after the curly", %{project: project} do
+      {:ok, [completion, _]} =
+        project
+        |> complete(my_module("%Project.Structs.Account{|}"))
+        |> fetch_completion(kind: :field)
+
+      assert completion.insert_text == "last_login_at: ${1:last_login_at}"
+      assert completion.insert_text_format == :snippet
+      assert completion.kind == :field
+      assert completion.label == "last_login_at: last_login_at"
+    end
+
+    test "should complete when after the comma", %{project: project} do
+      {:ok, [_, completion]} =
+        project
+        |> complete(my_module("%Project.Structs.Account{last_login_at: nil, |}"))
+        |> fetch_completion(kind: :field)
+
+      assert completion.insert_text == "user: ${1:user}"
+      assert completion.label == "user: user"
+    end
+
+    test "should complete even the struct module is aliased", %{project: project} do
+      source = ~q[
+        defmodule MyModule do
+          alias Project.Structs.Account, as: LocalAccount
+
+          def account(%LocalAccount{|} = account) do
+            account
+          end
+        end
+      ]
+
+      {:ok, [completion, _]} =
+        project
+        |> complete(source)
+        |> fetch_completion(kind: :field)
+
+      assert completion.insert_text == "last_login_at: ${1:last_login_at}"
+      assert completion.insert_text_format == :snippet
```

Comment:
maybe use `apply_completion` rather than asserting the snippet.

Reply:
`apply_completion` requires a `TextEdit{}`. I think a normal snippet would suffice here, as it is not a structural complement like ` %Strr|`, which modifies the position.